pkg/k8shandler/collection.go

```diff
@@ -501,7 +523,10 @@ func isFluentdDaemonsetDifferent(current *apps.DaemonSet, desired *apps.DaemonSe
 
 	different := false
 
-	// TODO: populate this
+	if isImageDifference(current, desired) {
+		logrus.Infof("Fluentd image change found, updating %q", current.Name)
+		return desired, true
```

Comment:
Why do we need to return `desired` here?

Reply:
I'm going to update this to set the image in current and return that later.